src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultImportDocumentManager.cs

```diff
@@ -61,7 +58,7 @@ public override void OnSubscribed(VisualStudioDocumentTracker tracker)
             foreach (var import in imports)
             {
                 var importFilePath = import.PhysicalPath;
-                Debug.Assert(importFilePath != null);
+                Assumes.NotNull(importFilePath);
```

Comment:
This seems odd. Can `PhyiscalPath` be annotated differently?

This would be a runtime change, where previously the `TryGetValue` check would (I assume) have returned false without issue, but now we'll throw an exception. I can't think of a time when I've seen the debug assert fail, but it might be safer for this to be `if (importFilePath is not null)`

Reply:
That case is interesting. As soon as I add `Debug.Assert` the compiler starts treating `importFilePath` as nullable and creates an error later in the code:
![devenv_7tryMKfPvo](https://user-images.githubusercontent.com/70431552/193471843-8a8679b3-d6d3-46ca-9245-c51a88cf52cc.gif)

Should I remove `Debug.Assert` at all or bring it back and suppress all compiler complaints with `!` (3 such suppressions will be required)?

Reply:
That partly makes sense, because once the compiler sees you check for null, it takes that as a sign that null is a potential value, but I don't know why it wouldn't complain about the null-oblivious local being passed to `TryGetValue` I guess when (if?) the compiler is annotated this will make sense again.

If you haven't worked it out already, I definitely prefer not putting `!` suppressions in, so I would put back the assert, but then just add `if (importFilePath is null) continue;` and call it a day. No runtime behaviour change, but also no suppressions to bite us later.